### PR TITLE
FollowUp cinema infrastructure

### PR DESCRIPTION
**Pull request by Frecherenkel60**
## Describe your changes

- added endpoint to add new cinema
- added endpoint to add cinemahall (with included seating plan)
- added endpoints to get cinemas, cinemahalls and seating plans
- refactored some code in the city service
- added tests

## Issue ticket number
follow-up on #14 (pr: #19)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Documentation added?

### CODE REVIEW

File: src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java [hunk 1]
```diff
@@ -1,20 +1,75 @@
...
+        for (int row = 1; row <= requestObject.getRows(); row++) {
+            // Loges are only in the last row, otherwise the halls offer regular seats
+            SeatType activeType = row == requestObject.getRows() ? logeType : regularType;
+
+            for (int place = 1; place <= requestObject.getPlacesPerRow(); place++) {
+                seats.add(new SeatBlueprint(newSeatingPlan, activeType, row, place));
+            }
+        }
+
+        newSeatingPlan.setSeats(seats);
```
Comment (Nico1808) [inline src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java:53]: Comment or other Method name

File: src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java [hunk 2]
```diff
@@ -1,20 +1,75 @@
...
+        newCinemaHall.setFloor(requestObject.getFloor());
+
+        return cinemaHallRepository.save(newCinemaHall);
+    }
+
+    public List<CinemaHall> getAllCinemaHallsByCinema(Cinema cinema) {
+        return cinemaHallRepository.findByCinema(cinema);
+    }
 
     public List<CinemaHall> getAllCinemaHalls() {
```
Comment (Nico1808) [inline src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java:67]: unneccessary

**Review by Nico1808 — APPROVED** [review]
Hasch super gemacht kevin

File: src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java (hunk 2 above)
Comment (Frecherenkel60) [reply src/main/java/com/wwi21sebgroup5/cinema/services/CinemaHallService.java]: replaced by getCinemaHallsByCinemaId

**Pull request merged**